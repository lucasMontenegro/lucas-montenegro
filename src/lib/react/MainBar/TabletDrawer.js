import { makeStyles } from "@material-ui/core/styles"
import useTranslation from "lib/react/useTranslation"
import React from "react"
import Drawer from "@material-ui/core/Drawer"
import List from "@material-ui/core/List"
import Tooltip from "@material-ui/core/Tooltip"
import ListItem from "@material-ui/core/ListItem"
import { useAppDrawerOpener } from "lib/react/AppDrawer"
//import { useDashboardOpener } from "lib/react/Dashboard"
import ListItemIcon from "@material-ui/core/ListItemIcon"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
import Divider from "@material-ui/core/Divider"
import TabletNav from "./TabletNav"
const useStyles = makeStyles(theme => ({
  drawerPaper: {
    width: theme.spacing(10) + 1,
  },
  muiLi: {
    justifyContent: `center`,
  },
  liIcon: {
    fontSize: theme.spacing(3),
    minWidth: `initial`,
  },
}), { name: `lib-react-main_bar-tablet_drawer` })
export default function TabletDrawer () {
  const classes = useStyles()
  const t = useTranslation()
  return (
    <Drawer classes={{ paper: classes.drawerPaper }} variant="permanent">
      <List disablePadding>
        <li>
          <Tooltip
            title={t({
              en: () => `Open dashboard`,
              es: () => `Abrir panel de control`,
            })}
          >
            <ListItem
              className={classes.muiLi}
              button
              aria-controls="lib-react-app_drawer"
              onClick={useAppDrawerOpener()}
            >
            {/*
              aria-controls="lib-react-dashboard"
              onClick={useDashboardOpener()}
            >
            */}
              <ListItemIcon className={classes.liIcon}>
                <FontAwesomeIcon icon={[`fas`, `bars`]} />
              </ListItemIcon>
            </ListItem>
          </Tooltip>
        </li>
        <Divider />
        <li><TabletNav t={t} /></li>
      </List>
    </Drawer>
  )
}