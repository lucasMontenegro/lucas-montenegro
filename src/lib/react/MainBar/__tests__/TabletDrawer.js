import renderer from "react-test-renderer"
import React from "react"
import TabletDrawer from "../TabletDrawer"
jest.mock(`@material-ui/core/styles`, () => ({
  __esModule: true,
  makeStyles: (styles, options) => () => {
    const classes = styles({ spacing: n => 8*n })
    return Object.keys(classes).reduce((obj, key) => {
      obj[key] = `${options.name}-${key}(${JSON.stringify(classes[key]).replace(/"/gm, `'`)})`
      return obj
    }, {})
  },
}))
jest.mock(`lib/react/useTranslation`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: () => source => (
      <div className="translation">
        {Object.keys(source).map(languageCode => (
          <div key={languageCode} className={languageCode}>{source[languageCode]()}</div>
        ))}
      </div>
    ),
  }
})
jest.mock(`@material-ui/core/Drawer`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="Drawer" /> }
})
jest.mock(`@material-ui/core/List`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="List" /> }
})
jest.mock(`@material-ui/core/Tooltip`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="Tooltip" /> }
})
jest.mock(`@material-ui/core/ListItem`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: props => (
      <div {...props} className={`ListItem ${props.className}`} onClick={props.onClick()} />
    ),
  }
})
jest.mock(`lib/react/Dashboard`, () => ({
  __esModule: true,
  useDashboardOpener: () => () => `useDashboardOpener()`,
}))
jest.mock(`@material-ui/core/ListItemIcon`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: props => <div {...props} className={`ListItemIcon ${props.className}`} />,
  }
})
jest.mock(`lib/react/fontAwesome`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    FontAwesomeIcon: props => <i {...props} className="FontAwesomeIcon" />,
  }
})
jest.mock(`@material-ui/core/Divider`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="Divider" /> }
})
jest.mock(`../TabletNav`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="TabletNav" /> }
})
describe(`../TabletDrawer`, () => {
  it(`should use the right versions of its dependencies`, () => {
    expect(jestUtils.getDependencies([`@material-ui/core`, `react`])).toMatchSnapshot()
  })
  describe(`<TabletDrawer />`, () => {
    it(`should render`, () => {
      expect(renderer.create(<TabletDrawer />).toJSON()).toMatchSnapshot()
    })
  })
})