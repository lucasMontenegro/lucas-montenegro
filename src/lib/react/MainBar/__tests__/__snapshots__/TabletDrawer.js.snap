// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../TabletDrawer <TabletDrawer /> should render 1`] = `
<div
  className="Drawer"
  classes={
    Object {
      "paper": "lib-react-main_bar-tablet_drawer-drawerPaper({'width':81})",
    }
  }
  variant="permanent"
>
  <div
    className="List"
    disablePadding={true}
  >
    <li>
      <div
        className="Tooltip"
        title={
          <div
            className="translation"
          >
            <div
              className="en"
            >
              Open dashboard
            </div>
            <div
              className="es"
            >
              Abrir panel de control
            </div>
          </div>
        }
      >
        <div
          aria-controls="lib-react-app_drawer"
          button={true}
          className="ListItem lib-react-main_bar-tablet_drawer-muiLi({'justifyContent':'center'})"
          onClick="useAppDrawerOpener()()"
        >
          <div
            className="ListItemIcon lib-react-main_bar-tablet_drawer-liIcon({'fontSize':24,'minWidth':'initial'})"
          >
            <i
              className="FontAwesomeIcon"
              icon={
                Array [
                  "fas",
                  "bars",
                ]
              }
            />
          </div>
        </div>
      </div>
    </li>
    <div
      className="Divider"
    />
    <li>
      <div
        className="TabletNav"
        t={[Function]}
      />
    </li>
  </div>
</div>
`;

exports[`../TabletDrawer should use the right versions of its dependencies 1`] = `
Object {
  "@material-ui/core": "^4.5.0",
  "react": "^16.10.1",
}
`;
